src/pgpsda_fmt_plug.c

```diff
@@ -0,0 +1,237 @@
+/*
+ * Format for brute-forcing PGP SDAs (self-decrypting archives).
+ *
+ * This software is Copyright (c) 2017, Dhiru Kholia <dhiru.kholia at gmail.com>,
+ * and it is hereby released to the general public under the following terms:
+ *
+ * Redistribution and use in source and binary forms, with or without modification,
+ * are permitted.
+ */
+
+#if FMT_EXTERNS_H
+extern struct fmt_main fmt_pgpsda;
+#elif FMT_REGISTERS_H
+john_register_one(&fmt_pgpsda);
+#else
+
+#include <string.h>
+#include <openssl/cast.h>
+
+#include "arch.h"
+#include "misc.h"
+#include "memory.h"
+#include "common.h"
+#include "formats.h"
+#include "johnswap.h"
+#ifdef _OPENMP
+#include <omp.h>
+#ifndef OMP_SCALE
+#define OMP_SCALE               1 // this is a slow format
+#endif
+#endif
+#include "sha.h"
+#include "loader.h"
+#include "pgpsda_common.h"
+#include "memdbg.h"
+
+#define FORMAT_LABEL            "pgpsda"
+#define FORMAT_NAME             ""
+#define ALGORITHM_NAME          "PGP SDA SHA1 " ARCH_BITS_STR
+#define PLAINTEXT_LENGTH        125
+#define SALT_SIZE               sizeof(struct custom_salt)
+#define SALT_ALIGN              sizeof(int)
+#define BINARY_SIZE             8
+#define BINARY_ALIGN            sizeof(uint32_t)
+#define BENCHMARK_COMMENT       ""
+#define BENCHMARK_LENGTH        -1
+#define MIN_KEYS_PER_CRYPT      1
+#define MAX_KEYS_PER_CRYPT      1
+#define FORMAT_TAG              "$pgpsda$"
+#define FORMAT_TAG_LENGTH       (sizeof(FORMAT_TAG) - 1)
+
+static struct custom_salt *cur_salt;
+
+static char (*saved_key)[PLAINTEXT_LENGTH + 2];
```

Comment:
Why do you keep defining saved_key one byte too large?
copy&paste error?

I just replaced all the (*saved_key)[PLAINTEXT_LENGTH + 2] with (*saved_key)[PLAINTEXT_LENGTH + 1], and didn't find any problems. Not even in pfx_fmt_plug.c which had a comment that +2 is needed for some unknown reason.

Reply:
Yes, it is a copy-and-paste error. Thanks for catching it!